City-of-Helsinki/helsinki-paatokset
# PP-366: Install news feed component

**Setup environment**
- Make sure you have this line in your `.env.local` file: `REACT_APP_ELASTIC_URL=https://paatokset-proxy-test.agw.arodevtest.hel.fi/`
  - If you don't add it and run `make up` 
- Checkout branch and run `make new`
- Make sure you have a `public/sites/default/local.services.yml` file with this content:
```
parameters:
  helfi_api_base.logger_enabled: false
  helfi_api_base.internal_domains:
    - 'hel.fi'
    - 'www.hel.fi'
    - 'paatokset.hel.fi'
    - 'avustukset.hel.fi'
    - 'helsinki-paatokset.docker.so'
  twig.config:
    debug: true
    auto_reload: true
    cache: false
```
- Login to the container with `make shell` and run the following commands:
```
drush migrate-import ahjo_decisionmakers:all && drush migrate-import ahjo_decisionmakers:all_sv
drush migrate-import ahjo_trustees:council
drush migrate-import ahjo_meetings:latest
drush ap:update meetings 02900202220 -v
drush migrate-import ahjo_decisions:latest
drush ahjo-proxy:get-motions -v
drush ahjo-proxy:update-decisions -v
drush queue:run ahjo_api_subscriber_queue -v
drush ahjo-proxy:update-decisions --update -v
```
**Create the following pages:**
- Landing page: `/etusivu`
  - Add a Hero-element to the page with the "Search bar" layout
  - Add a "News list" component to the page with "Päätökset" as a filter tag
  - Create swedish and english translations
- Landing page: `/paattajat`
  - Add to the main navigation
  - Create swedish and english translations: `/decisionmakers` and `/beslutsfattare`
- Landing page: `/evasteasetukset`
  - Add some content components to the page. After saving they should appear above the cookie consent element
- Landing page: `/kokouskalenteri` 
  - Add to the main navigation
  - Create swedish and english translations: `/meetings-calendar` and `/moteskalender`
- Page: `/tietoa-paatoksenteosta` 
  - Add to the main navigation
  - Add some content components to the page
    - Sidebar content should show in the sidebar, upper and main content components should show before the diagram and lower content area components should show up below the sidebar and main content area 
- Page: "Kuka päättää ja miksi"
  - Add to the main navigation under `tietoa-paatoksenteosta`
  - Add some content components to the page
- Page: `/kuulutukset-ja-ilmoitukset`
  - Add to main navigation
- Page: "Mitä ovat kuulutukset, ilmoitukset ja julkipanot?"
  - Add to the main navigation under `kuulutukset-ja-ilmoitukset`
- Add a few articles and make sure some of them share the same keywords
  - They should be listed on the `kuulutukset-ja-ilmoitukset` page
  - After clicking the keyword tag, you should get to a page that lists all articles with that keywrod

**To test:**
- Open this page: https://helsinki-paatokset.docker.so/fi/health , it should print a timestamp
  - Put the site into maintenance mode: https://helsinki-paatokset.docker.so/fi/admin/config/development/maintenance
  - Open the frontpage in a private browsing window. You should get a "Päivitämme sivustoa tällä hetkellä" page with a 503 HTTP response
  - Reload the `/health` page again and also open it in a private browsing window. It should still print the timestamp and respond with a 200 HTTP code.
- Open the page in a private browsing window. The cookie popup should appear. Click "Lisätietoja"
  - Only accept essential cookies from the popup. The page should reload. Scroll through the list and make sure none of the other cookie categories are checked
  - At the bottom of the page you should see a banner that says something like "Sivun palautenappeja ei voida näyttää"
  - Accept all cookies. The feedback banner should appear now
  - Try selecting and unselecting different cookie categories and then saving the changes. The selections you made should stick after saving and reloading
- Open the frontpage and test the search application. It should give you some suggestions. After making a selection and clicking "Search / Hae" it should redirect you to the decisions search page
- Check that the meetings calendar page works and loads meetings: https://helsinki-paatokset.docker.so/fi/kokouskalenteri  
  - Scroll back to november and check that the meeting links work
- Check that the decisionmakers list works: https://helsinki-paatokset.docker.so/fi/paattajat
  - Check that the search on this page works (the results come from the test site, so their links might not be correct)
  - Open the swedish version of this page. The search should only suggest and display swedish decisionmaker titles 
  - Open a few decisionmakers from each category and check that their pages and subpages work (most won't have any decisions or documents)
  - Make sure you can switch between language versions even if the nodes aren't translated
  - Open a few council member pages and theck that their page layout is OK.
  - Open https://helsinki-paatokset.docker.so/fi/paattajat/exlhilkkaahde and check that the initiatives etc accordions are present and work
- Check some decisions and make sure the layout works properly: https://helsinki-paatokset.docker.so/fi/admin/content/decisions
  - Make sure you can switch between language versions
  - Check that the dropdown selector and arrows works on this page: https://helsinki-paatokset.docker.so/fi/asia/hel-2022-012471/098b18ae-e5ee-cb9c-81f4-842dbb800007?paatos=098b18ae-e5ee-cb9c-81f4-842dbb800007
